### ISSUE TITLE
[v4 css] form-control-sm styling

### ISSUE THREAD

**Issue by Mottie** [twbs/bootstrap#21587]
When comparing form elements with a "form-control" versus a "form-control-sm" class in Bootstrap v4.0.0-alpha.6, there are some notable differences &ndash; I'm using Windows 10 and Chrome version 55.0.2883.87 m (64-bit), in case that matters.

[demo](http://codepen.io/Mottie/pen/YNXxeL) (updated!)

![2017-01-06 19_38_24-bootstrap v4 0 0-alpha 6 issue](https://cloud.githubusercontent.com/assets/136959/21738259/d2b94ff2-d447-11e6-8390-ab2bd13bd887.png)

* <del>`select.form-control-sm` &ndash; border too dark</del>
* `input.form-control-sm`&ndash; mismatched height <del>& inset border</del>

-----

Update: sorry, I didn't realize that both classes needed to be applied.... the `input` height is still slightly off.

**Comment by sergiopantoja** [comment]
Same on macOS Sierra and Chrome version 55.0.2883.95 (64-bit).

![image](https://cloud.githubusercontent.com/assets/2530982/21737938/5eca2a30-d44c-11e6-952d-b8323d21adb6.png)

**Comment by sergiopantoja** [comment]
@Mottie 

So it looks like `.form-control-sm` needs to be used with `.form-control`. For example:
```html
<input class="form-control form-control-sm" type="text">
```
```html
<select class="form-control form-control-sm">
  <option>Small select</option>
</select>
```

Can you try this and see if you still encounter the same issue?

**Comment by Mottie** [comment]
Ahh, you're right!... but the input height is still not matching perfectly... here is a zoomed in screenshot:

![2017-01-06 19_38_24-bootstrap v4 0 0-alpha 6 issue](https://cloud.githubusercontent.com/assets/136959/21738259/d2b94ff2-d447-11e6-8390-ab2bd13bd887.png)

**Comment by mdo** [comment]
Fixed by #22287.
